presto-main/src/main/java/com/facebook/presto/operator/scalar/RowToJsonCast.java

```diff
@@ -72,19 +74,21 @@ public BuiltInScalarFunctionImplementation specialize(BoundVariables boundVariab
 
         List<Type> fieldTypes = type.getTypeParameters();
         List<JsonGeneratorWriter> fieldWriters = new ArrayList<>(fieldTypes.size());
+        List<TypeSignatureParameter> typeSignatureParameters = type.getTypeSignature().getParameters();
+        List<String> fieldNames = new ArrayList<>(fieldTypes.size());
         for (int i = 0; i < fieldTypes.size(); i++) {
+            fieldNames.add(typeSignatureParameters.get(i).getNamedTypeSignature().getName().orElse(""));
```

Comment:
Do we check whether or not typeSignatureParameters has the same or large size of fieldTypes.size()? Otherwise could we get out of range?

Reply:
For rows, typeSignature and typeParameters are both transformation of the fields so they will have the same size

Please see: https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/type/RowType.java

Reply:
Do we want to add assert that the two size are same?

Reply:
Done